src/main/java/org/openstreetmap/atlas/geography/boundary/CountryBoundaryMapArchiver.java

```diff
@@ -19,42 +27,119 @@
+            File::new, Optionality.REQUIRED);
+
+    // Options
+    protected static final Switch<Rectangle> BOUNDS = new Switch<>("bounds", "The bounds",
+            Rectangle::forString, Optionality.OPTIONAL, Rectangle.MAXIMUM.toCompactString());
+    protected static final Switch<Boolean> CREATE_SPATIAL_INDEX = new Switch<>("createSpatialIndex",
             "Indicator whether to create a spatial grid index and include that in the output.",
             Boolean::parseBoolean, Optionality.OPTIONAL, Boolean.FALSE.toString());
+    protected static final Switch<Integer> OCEAN_BOUNDARY_ZOOM_LEVEL = new Switch<>(
+            "oceanBoundaryZoomLevel",
+            "The zoom level at which to create ocean tiles to fill in potential voids. Recommended value: 3",
+            Integer::parseInt, Optionality.OPTIONAL);
+    protected static final Switch<Boolean> SAVE_GEOJSON_WKT = new Switch<>("saveGeojsonWkt",
+            "Save the country boundaries to Geojson and WKT", Boolean::parseBoolean,
+            Optionality.OPTIONAL, Boolean.FALSE.toString());
+
+    // Internal
+    private static final double JTS_SNAP_PRECISION = .000000000000001;
 
     public static void main(final String[] args)
     {
         new CountryBoundaryMapArchiver().run(args);
     }
 
-    /**
-     * @param resource
-     *            The {@link Resource} to read the {@link CountryBoundaryMap} from
-     * @return the created {@link CountryBoundaryMap}
-     */
-    public CountryBoundaryMap read(final Resource resource)
+    protected CountryBoundaryMap generateOceanBoundaryMap(final CountryBoundaryMap boundaryMap,
+            final Iterable<SlippyTile> allTiles)
+    {
+        final CountryBoundaryMap finalBoundaryMap = new CountryBoundaryMap();
+        int oceanCountryCount = 0;
+        // add all ocean boundaries to the new boundary map
+        logger.info("Calculating ocean boundaries...");
+        for (final SlippyTile tile : allTiles)
+        {
+            final Time start = Time.now();
+            final String countryCode = String.format("O%02d", oceanCountryCount);
+            final Geometry countryGeometry = geometryForShard(tile.bounds(), boundaryMap);
+            if (!countryGeometry.isEmpty())
+            {
+                if (countryGeometry instanceof Polygon)
+                {
+                    finalBoundaryMap.addCountry(countryCode, (Polygon) countryGeometry);
+                }
+                if (countryGeometry instanceof MultiPolygon)
+                {
+                    finalBoundaryMap.addCountry(countryCode, (MultiPolygon) countryGeometry);
+                }
+                logger.info("Added Ocean Country {} in {}", countryCode, start.elapsedSince());
+            }
+            else
+            {
+                logger.info("Skipped Ocean Country {} in {}. It is land covered.", countryCode,
+                        start.elapsedSince());
+            }
+            oceanCountryCount++;
```

Comment:
I used to increment this only if the boundary existed, so our ocean boundaries will have different names. For simplicity could we move this increment back to the if statement?